packages/framework/tests/Unit/Model/Order/Item/OrderItemTest.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Model\Order\Item;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Model\Order\Item\Exception\MainVariantCannotBeOrderedException;
+use Shopsys\FrameworkBundle\Model\Order\Item\Exception\WrongItemTypeException;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItem;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData;
+use Shopsys\FrameworkBundle\Model\Order\Order;
+use Shopsys\FrameworkBundle\Model\Payment\Payment;
+use Shopsys\FrameworkBundle\Model\Pricing\Price;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductData;
+use Shopsys\FrameworkBundle\Model\Transport\Transport;
+
+class OrderItemTest extends TestCase
+{
+    public function testTransportCannotBeSetForWrongType(): void
+    {
+        $orderItem = $this->createOrderPayment();
+
+        $this->expectException(WrongItemTypeException::class);
+        $orderItem->setTransport($this->createTransportMock());
+    }
+
+    public function testTransportCannotBeGottenFromWrongType(): void
+    {
+        $orderItem = $this->createOrderPayment();
+
+        $this->expectException(WrongItemTypeException::class);
+        $orderItem->getTransport();
+    }
+
+    public function testEditTransportTypeEditsTransport(): void
+    {
+        $orderItem = $this->createOrderTransport();
+
+        $orderItemData = new OrderItemData();
+        $transport = $this->createTransportMock();
+        $orderItemData->transport = $transport;
+        $orderItem->edit($orderItemData);
+
+        $this->assertSame($transport, $orderItem->getTransport());
+    }
+
+    public function testPaymentCannotBeSettedForWrongType(): void
+    {
+        $orderItem = $this->createOrderProduct();
+
+        $this->expectException(WrongItemTypeException::class);
+        $orderItem->setPayment($this->createPaymentMock());
+    }
+
+    public function testPaymentCannotBeGettedFromWrongType(): void
```

Comment:
The past participle of `get` is `got` or `gotten`, not `getted` (`taken` can be used as well and is more common in such context).